sdk/go/auto/local_workspace.go

```diff
@@ -779,7 +779,24 @@ func (l *LocalWorkspace) Install(ctx context.Context, opts *InstallOptions) erro
 	if opts != nil && opts.Stderr != nil {
 		stderrWriters = append(stderrWriters, opts.Stderr)
 	}
-	stdout, stderr, errCode, err := l.runPulumiInputCmdSync(ctx, nil, stdoutWriters, stderrWriters, "install")
+	args := []string{"install"}
+	// Pulumi 3.130.0 introduced the `--use-language-version-tools` flag. If we are on a version
+	// before that, ignore the flag.
```

Comment:
I think usually we just didn't do anything like this, and just let automation API fail if the flag does not exist in the pulumi version the user is using (for `--continue-on-error` for example).  

I wonder if it would make sense to just error out here instead of silently not doing what the user requested?

Reply:
Yes, I think I agree with you. We have for example

```go
func (l *LocalWorkspace) AddEnvironments(ctx context.Context, stackName string, envs ...string) error {
	// 3.95 added this command (https://github.com/pulumi/pulumi/releases/tag/v3.95.0)
	if l.pulumiCommand.Version().LT(semver.Version{Major: 3, Minor: 95}) {
		return errors.New("AddEnvironments requires Pulumi CLI version >= 3.95.0")
	}
    ...
```

The caller will have to guard against this, which I think is better than making options silently do nothing.